core/src/main/java/org/apache/accumulo/core/gc/ReferenceScan.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.gc;
+
+import org.apache.accumulo.core.data.TableId;
+
+/**
+ * A GC reference to a Scan entry.
+ */
+public class ReferenceScan extends ReferenceFile {
+
+  public ReferenceScan(TableId tableId, String metaDataEntry) {
+    super(tableId, metaDataEntry);
+  }
+
+  @Override
+  public boolean isScan() {
+    return true;
+  }
```

Comment:
I'm not really sure the method and the boolean really buy us much here. The truth value of the method is basically synonymous with `instanceof ReferenceScan`, so it seems a bit redundant. I would probably choose either one or the other. The subclass variant might also have issues with regards to the fact that it doesn't override equals and hashCode, which are defined in ReferenceFile. That could be a problem, because you can get compareTo returning 0, even though they are not equal. I'm sure there are other similar problems like this, too.